### DISCUSSION TITLE
Where can I find documentaiton for clipper API?

### DISCUSSION THREAD

**Discussion by piranha32** [TriliumNext/Trilium#4332] [Q&A] [answered]
I would like to use Trilium to automatically record results from my experiments, and the clipping interface, as used by the web clipper plugins would work great for me. By looking at the requests from the plugin, I can see that it calls methods from /api/clipper api.
Unfortunately, I can't see it documented in the ETAPI specification. Is there a document, or an example, which shows how to use this endpoint?

**Comment by FliegendeWurst** [comment]
Last I checked there was no documentation. So you have two options:

1. Read https://github.com/zadam/trilium-web-clipper/blob/master/trilium_server_facade.js and associated files
2. Observe the requests the Web Clipper extension makes using the browser dev tools

**Reply by piranha32** [reply]
This is where I started. The plugin code is a maze of callbacks, and the packet dump from communication with the server, with some requests repeated twice, made little sense.
Unfortunately, requests sent by the plugin don't show up in the dev tools console in the log of the requests made by the page from which I was making clipping. 
I'll have to dig into plugin development and debugging, or organize the experiment logs slightly differently, what doesn't feel like a very bad idea.

**Reply by FliegendeWurst** [reply]
> Unfortunately, requests sent by the plugin don't show up in the dev tools console in the log of the requests made by the page from which I was making clipping.

Yes, to see those you need to use the "Browser Toolbox" if you're running Firefox (Chrome may offer something similar).

Last I checked (3 years ago) the API was fairly simple though:
1. Send a request to /api/login/token, json body: username and password. Get the token from the json response.
2. Send a request to /api/clipper/notes with json data like `{"title":"line 1","content":"<p>line 2</p><p>line 3</p>","clipType":"note"}`, and an Authorization header that contains the token.

**Reply by piranha32** [reply]
Thanks, I didn't have it enabled. Seeing the requests on browser level should make much more sense.

**Comment by zadam** [comment, marked as answer]
Clipper API is not documented since it's not considered to be "public API". You can still use it, but there are no guarantees on the stability etc. (even though major changes are unlikely)
